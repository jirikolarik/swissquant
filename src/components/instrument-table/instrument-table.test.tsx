import { shallow } from 'enzyme'
import React from 'react'
import InstrumentTable from '../instrument-table'

const dummyData = [
  {
    instrumentId: 70,
    timeSeries: {
      entries: [
        {
          d: "2010-01-01",
          n: 30,
          v: 6545.91
        },
        {
          d: "2010-01-04",
          n: 12,
          v: 6631.44
        },
      ]
    }
  },   {
    instrumentId: 73,
    timeSeries: {
      entries: [
        {
          d: "2010-01-01",
          n: 50,
          v: 645.91
        },
        {
          d: "2010-01-04",
          n: 42,
          v: 631.44
        },
      ]
    }
  }
]

describe('Instrument table', () => {
  it('snapshot', () => {
    const wrapper = shallow(<InstrumentTable instruments={dummyData}  />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders all data', () => {
    const wrapper = shallow(<InstrumentTable instruments={dummyData}  />);

    expect(wrapper.text()).toEqual(expect.stringContaining('2010-01-01'));
    expect(wrapper.text()).toEqual(expect.stringContaining('2010-01-04'));
    expect(wrapper.text()).toEqual(expect.stringContaining('6545.91'));
    expect(wrapper.text()).toEqual(expect.stringContaining('6631.44'));
    expect(wrapper.text()).toEqual(expect.stringContaining('645.91'));
    expect(wrapper.text()).toEqual(expect.stringContaining('631.44'));
  });
});
