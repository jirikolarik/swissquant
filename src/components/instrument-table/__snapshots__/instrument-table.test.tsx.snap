// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instrument table snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InstrumentTable
    instruments={
        Array [
            Object {
              "instrumentId": 70,
              "timeSeries": Object {
                "entries": Array [
                  Object {
                    "d": "2010-01-01",
                    "n": 30,
                    "v": 6545.91,
                  },
                  Object {
                    "d": "2010-01-04",
                    "n": 12,
                    "v": 6631.44,
                  },
                ],
              },
            },
            Object {
              "instrumentId": 73,
              "timeSeries": Object {
                "entries": Array [
                  Object {
                    "d": "2010-01-01",
                    "n": 50,
                    "v": 645.91,
                  },
                  Object {
                    "d": "2010-01-04",
                    "n": 42,
                    "v": 631.44,
                  },
                ],
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="c--instrument-table--header"
>
          <div
                    className="c--instrument-table--header--item"
          >
                    Date
          </div>
          <div
                    className="c--instrument-table--header--item"
          >
                    2010-01-01
          </div>
          <div
                    className="c--instrument-table--header--item"
          >
                    2010-01-04
          </div>
</div>,
        Array [
          <div
            className="c--instrument-table--column"
>
            <div
                        className="c--instrument-table--header--item"
            >
                        70
            </div>
            <div
                        className="c--instrument-table--item"
                        onMouseOver={[Function]}
            >
                        6545.91
            </div>
            <div
                        className="c--instrument-table--item"
                        onMouseOver={[Function]}
            >
                        6631.44
            </div>
</div>,
          <div
            className="c--instrument-table--column"
>
            <div
                        className="c--instrument-table--header--item"
            >
                        73
            </div>
            <div
                        className="c--instrument-table--item"
                        onMouseOver={[Function]}
            >
                        645.91
            </div>
            <div
                        className="c--instrument-table--item"
                        onMouseOver={[Function]}
            >
                        631.44
            </div>
</div>,
        ],
      ],
      "className": "c--instrument-table--table",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="c--instrument-table--header--item"
>
              Date
</div>,
            Array [
              <div
                className="c--instrument-table--header--item"
>
                2010-01-01
</div>,
              <div
                className="c--instrument-table--header--item"
>
                2010-01-04
</div>,
            ],
          ],
          "className": "c--instrument-table--header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Date",
              "className": "c--instrument-table--header--item",
            },
            "ref": null,
            "rendered": "Date",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2010-01-01",
            "nodeType": "host",
            "props": Object {
              "children": "2010-01-01",
              "className": "c--instrument-table--header--item",
            },
            "ref": null,
            "rendered": "2010-01-01",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2010-01-04",
            "nodeType": "host",
            "props": Object {
              "children": "2010-01-04",
              "className": "c--instrument-table--header--item",
            },
            "ref": null,
            "rendered": "2010-01-04",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "70",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="c--instrument-table--header--item"
>
              70
</div>,
            Array [
              <div
                className="c--instrument-table--item"
                onMouseOver={[Function]}
>
                6545.91
</div>,
              <div
                className="c--instrument-table--item"
                onMouseOver={[Function]}
>
                6631.44
</div>,
            ],
          ],
          "className": "c--instrument-table--column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 70,
              "className": "c--instrument-table--header--item",
            },
            "ref": null,
            "rendered": 70,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2010-01-01",
            "nodeType": "host",
            "props": Object {
              "children": 6545.91,
              "className": "c--instrument-table--item",
              "onMouseOver": [Function],
            },
            "ref": null,
            "rendered": 6545.91,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2010-01-04",
            "nodeType": "host",
            "props": Object {
              "children": 6631.44,
              "className": "c--instrument-table--item",
              "onMouseOver": [Function],
            },
            "ref": null,
            "rendered": 6631.44,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "73",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="c--instrument-table--header--item"
>
              73
</div>,
            Array [
              <div
                className="c--instrument-table--item"
                onMouseOver={[Function]}
>
                645.91
</div>,
              <div
                className="c--instrument-table--item"
                onMouseOver={[Function]}
>
                631.44
</div>,
            ],
          ],
          "className": "c--instrument-table--column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 73,
              "className": "c--instrument-table--header--item",
            },
            "ref": null,
            "rendered": 73,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2010-01-01",
            "nodeType": "host",
            "props": Object {
              "children": 645.91,
              "className": "c--instrument-table--item",
              "onMouseOver": [Function],
            },
            "ref": null,
            "rendered": 645.91,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2010-01-04",
            "nodeType": "host",
            "props": Object {
              "children": 631.44,
              "className": "c--instrument-table--item",
              "onMouseOver": [Function],
            },
            "ref": null,
            "rendered": 631.44,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c--instrument-table--header"
>
            <div
                        className="c--instrument-table--header--item"
            >
                        Date
            </div>
            <div
                        className="c--instrument-table--header--item"
            >
                        2010-01-01
            </div>
            <div
                        className="c--instrument-table--header--item"
            >
                        2010-01-04
            </div>
</div>,
          Array [
            <div
              className="c--instrument-table--column"
>
              <div
                            className="c--instrument-table--header--item"
              >
                            70
              </div>
              <div
                            className="c--instrument-table--item"
                            onMouseOver={[Function]}
              >
                            6545.91
              </div>
              <div
                            className="c--instrument-table--item"
                            onMouseOver={[Function]}
              >
                            6631.44
              </div>
</div>,
            <div
              className="c--instrument-table--column"
>
              <div
                            className="c--instrument-table--header--item"
              >
                            73
              </div>
              <div
                            className="c--instrument-table--item"
                            onMouseOver={[Function]}
              >
                            645.91
              </div>
              <div
                            className="c--instrument-table--item"
                            onMouseOver={[Function]}
              >
                            631.44
              </div>
</div>,
          ],
        ],
        "className": "c--instrument-table--table",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="c--instrument-table--header--item"
>
                Date
</div>,
              Array [
                <div
                  className="c--instrument-table--header--item"
>
                  2010-01-01
</div>,
                <div
                  className="c--instrument-table--header--item"
>
                  2010-01-04
</div>,
              ],
            ],
            "className": "c--instrument-table--header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Date",
                "className": "c--instrument-table--header--item",
              },
              "ref": null,
              "rendered": "Date",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2010-01-01",
              "nodeType": "host",
              "props": Object {
                "children": "2010-01-01",
                "className": "c--instrument-table--header--item",
              },
              "ref": null,
              "rendered": "2010-01-01",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2010-01-04",
              "nodeType": "host",
              "props": Object {
                "children": "2010-01-04",
                "className": "c--instrument-table--header--item",
              },
              "ref": null,
              "rendered": "2010-01-04",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "70",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="c--instrument-table--header--item"
>
                70
</div>,
              Array [
                <div
                  className="c--instrument-table--item"
                  onMouseOver={[Function]}
>
                  6545.91
</div>,
                <div
                  className="c--instrument-table--item"
                  onMouseOver={[Function]}
>
                  6631.44
</div>,
              ],
            ],
            "className": "c--instrument-table--column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 70,
                "className": "c--instrument-table--header--item",
              },
              "ref": null,
              "rendered": 70,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2010-01-01",
              "nodeType": "host",
              "props": Object {
                "children": 6545.91,
                "className": "c--instrument-table--item",
                "onMouseOver": [Function],
              },
              "ref": null,
              "rendered": 6545.91,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2010-01-04",
              "nodeType": "host",
              "props": Object {
                "children": 6631.44,
                "className": "c--instrument-table--item",
                "onMouseOver": [Function],
              },
              "ref": null,
              "rendered": 6631.44,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "73",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="c--instrument-table--header--item"
>
                73
</div>,
              Array [
                <div
                  className="c--instrument-table--item"
                  onMouseOver={[Function]}
>
                  645.91
</div>,
                <div
                  className="c--instrument-table--item"
                  onMouseOver={[Function]}
>
                  631.44
</div>,
              ],
            ],
            "className": "c--instrument-table--column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 73,
                "className": "c--instrument-table--header--item",
              },
              "ref": null,
              "rendered": 73,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2010-01-01",
              "nodeType": "host",
              "props": Object {
                "children": 645.91,
                "className": "c--instrument-table--item",
                "onMouseOver": [Function],
              },
              "ref": null,
              "rendered": 645.91,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2010-01-04",
              "nodeType": "host",
              "props": Object {
                "children": 631.44,
                "className": "c--instrument-table--item",
                "onMouseOver": [Function],
              },
              "ref": null,
              "rendered": 631.44,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
